corehq/apps/app_manager/models.py

```diff
@@ -1025,14 +1034,14 @@ def validate_form(self):
                     "validation_problems": e.validation_problems,
                     "version": e.version,
                 }
-                vc = self.validation_cache = json.dumps(validation_dict)
+                vc = self.set_validation_cache(json.dumps(validation_dict))
             else:
-                vc = self.validation_cache = ""
+                vc = self.set_validation_cache("")
```

Comment:
Probably very simple but I am confused how this is resolving the issue. Can you share how this is working now?

Reply:
Previously, this line would fail for shadow forms because `validation_cache` wasn't a string but was instead a function that returned the parent's cache. Now `validation_cache` is always a string, but instead of code directly accessing it, it's accessed through `get_validation_cache`, `set_validation_cache`, and `clear_validation_cache`, which determine whether to get/set the form's own cache (for normal forms) or the parent's cache (for shadow forms).

Reply:
Thank you. 👍